### PR TITLE
Randomise wiring, add examine wire, examining wires scales off MEC

### PR DESCRIPTION
**Pull request by Kegdo**
## About The Pull Request

All machinery(code/datums/wires/) has randomised wiring.
You can examine wires.
If your MEC is >= 40, you can figure out what the wire does.
image

## Why It's Good For The Game
Now MEC is actually useful besides gun crafting. ~~Come on, lynch me, I dare you.~~

This was in FortuneMaster's design doc. Also known as "Rope Bunny Boy" on discord.

## Changelog
:cl:
tweak: all wires in machinery are randomised now.
add: you can now examine(MEC >= 40) wires to figure out what each does.
/:cl:

### CODE REVIEW

**Comment by Jan-Blasiak** [comment]
Do you mean all wires in CATEGORY (all airlocks, vendors), or all wires IN ALL MACHINES (every machine randomized)?

File: code/datums/wires/wires.dm [hunk 1]
```diff
@@ -71,15 +71,15 @@ var/list/wireColours = list("red", "blue", "green", "darkred", "orange", "brown"
 		src.wires[colour] = index
 		//wires = shuffle(wires)
 
-/datum/wires/proc/examine(index, mob/user)
+/datum/wires/proc/examine(index, mob/user,)
```
Comment (TheShown911) [inline code/datums/wires/wires.dm:74]: Guh?

File: code/datums/wires/wires.dm [hunk 2]
```diff
@@ -71,15 +71,15 @@ var/list/wireColours = list("red", "blue", "green", "darkred", "orange", "brown"
...
+	var/mec_stat = user.stats.getStat(STAT_MEC)
 
 	var/datum/wire_description/wd = get_description(index)
 	if(!wd)
 		return
-	//TODO: Port bay wires fully and integrate eris' skill system
-	//if(wd.skill_level && !user.skill_check(SKILL_ELECTRICAL, wd.skill_level))
-		//return
+	if(wd.skill_level > mec_stat)
+		return
```
Comment (TheShown911) [inline code/datums/wires/wires.dm:82]: Make it percentage based, as in 20/40 skill is 50% chance to get the right wire description and 50% chance to get a random wire description instead. That would be funny, I think.

Comment (Kegdo) [reply code/datums/wires/wires.dm:82]: Nah, then you'd be able to just figure out all the wires eventually, or never be sure the examine is correct
Anyway, bad idea, github mobile is lagging too muvh for me to make a better argument

File: code/datums/wires/airlock.dm [hunk 3]
```diff
@@ -1,13 +1,26 @@
 // Wires for airlocks
 
 /datum/wires/airlock/secure
-	random = 1
```
Comment (SirRichardFrancis) [inline code/datums/wires/airlock.dm:4]: Bruh.

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]
Awesome feature. But implementation is silly. A button for each wire? Cringe! Just make a description appear under each wire or something.
Better yet, add _short_ description of what wire do (main power, light, bolts, ai control, et cetera) before or right after a color string.

**Comment by Kegdo** [comment]
>Just make a description appear under each wire or something. 
 
 sounds easier than it sounds, the desc under each wire will BLOAT THE FUCK OUT of the wiring window. And I am fairly sure it will also bloat the code.
 
>Better yet, add _short_ description of what wire do (main power, light, bolts, ai control, et cetera) before or right after a color string.

same thing. Nah. The (stolen) bay examination code works best.

**Comment by SirRichardFrancis** [comment]
> sounds easier than it sounds, the desc under each wire will BLOAT THE FUCK OUT of the wiring window. And I am fairly sure it will also bloat the code.
> same thing. Nah. The (stolen) bay examination code works best.

It _is_ very easy. Almost the same manipulations as with adding a button.

**Comment by Kegdo** [comment]
shush, thot
_i just really don't want to rework this anymore, it works so don't touch it_

**Comment by SirRichardFrancis** [comment]
> shush, thot _i just really don't want to rework this anymore, it works so don't touch it_ (and it WILL bloat the menu)

It's just bad implementation. Short description of wire's function will not bloat anything, it will take no more space than "examine" button does.

**Comment by Kegdo** [comment]
image

**Review by drexample — APPROVED** [review]

**Pull request merged**